src/main/java/net/sf/jabref/gui/EntryTypeList.java

```diff
@@ -127,16 +126,6 @@ public void addDefaultActionListener(ActionListener l) {
     }
 
     @Override
-    public void actionPerformed(ActionEvent e) {
-        // Default button pressed.
-        if (e.getSource() == def) {
-            def.setEnabled(false);
-        } else {
-            super.actionPerformed(e);
```

Comment:
If I read the code correctly, the propagation of the event to super gets lost in your refactoring. Please make sure that you preserve the existing functionality, unless it is a bug.